client/blocks/payment-request/hooks.js

```diff
@@ -0,0 +1,356 @@
+/* global wc_stripe_payment_request_params */
+
+/**
+ * External dependencies
+ */
+import { useState, useEffect, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import $ from 'jquery';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getCartDetails,
+	createPaymentRequest,
+	updateShippingOptions,
+	updateShippingDetails,
+	createOrder,
+} from '../../api';
+import { updatePaymentRequest } from '../stripe-utils';
+
+/**
+ * This hook takes care of creating a payment request and making sure
+ * you can pay through said payment request.
+ *
+ * @param {Object} stripe The stripe object used to create the payment request.
+ *
+ * @return {Array} An array; first element is the payment request; second element is the payment
+ *                 requests type.
+ */
+export const usePaymentRequest = ( stripe ) => {
+	const [ paymentRequest, setPaymentRequest ] = useState( null );
+	const [ paymentRequestType, setPaymentRequestType ] = useState( null );
+	useEffect( () => {
+		// Do nothing if Stripe object isn't loaded or paymentRequest already exists.
+		if ( ! stripe || paymentRequest ) {
+			return;
+		}
+
+		getCartDetails().then( ( cart ) => {
```

Comment:
This is really only run once the site loads and the payment request is created. When you then click the payment request button, the payment request is updated with new information from the cart 🙂 

I think that logic wasn't pushed to GitHub when you were reviewing this, so you probably need to take another look to see that part of the logic 🤔  I'll edit this part of the message to include a link to that code. ⌛

Reply:
Ah yea, I forgot to think about that!

I've added this effect to the `usePaymentRequest` hook (the one that creates the payment request) in 87940a9cc65f12f208319fbec3a0e226b8319fc4.
I decided not to create a new hook since this effect seems like a necessary function of managing the payment request. Let me know if you think it should be its own hook 🤔

Reply:
No, it's better the way you did it.